app/components/pilas-blockly.js

```diff
@@ -389,8 +395,20 @@ export default Component.extend({
   },
 
   showExpectationFeedback() {
-    this.get('failedExpects').forEach(({ declaration, expect }, i) =>
-      addWarning(declarationWithName(declaration), expect, -i)// TODO: Add priority?
+    // Order is important. Warnings should be added first. This way, if errors appear, warning bubbles will be painted red.
+    this.expectationFeedbackFor(
+      this.get('failedExpects').filter(fe => !isCritical(fe)),
+      addWarning
+    )
+    this.expectationFeedbackFor(
+      this.get('failedExpects').filter(isCritical),
+      addError
+    )
+  },
+
+  expectationFeedbackFor(failedExceptations, addFeedback) {
+    failedExceptations.forEach(({ declaration, description }, i) =>
+      addFeedback(declarationWithName(declaration), description, -i)// TODO: Add priority?
     )
   },
```

Comment:
Esto está medio feo, pero creo que hacerlo mejor podría implicar bastantes cambios. Pero si se les ocurre algo me avisan.

Reply:
- expectationFeedbackFor debería quizás tener un nombre imperativo ¿Agregarle show adelante?
- `this.get('failedExpects').filter(isCritical)` debería estar en el procedimiento. Los parámetros deberían ser la función addError/addWarning y un booleano.

Reply:
Me gustaría poder componer la negación (? Done!